File: apps/backend/app/schemas/market_insight.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class BestsellerBookResponse(BaseModel):
    id: UUID
    rank: int
    title: str
    author: str | None
    publisher: str | None
    category: str | None
    sales_volume: int | None

    model_config = ConfigDict(from_attributes=True)


class BestsellerThemeResponse(BaseModel):
    category: str
    total_volume: int
    book_count: int


class MarketInsightsResponse(BaseModel):
    source: str = "publishnews"
    period_label: str | None
    last_updated: datetime | None
    books: list[BestsellerBookResponse]
    themes: list[BestsellerThemeResponse]

File: apps/frontend/types/market-insight.ts
export type BestsellerBook = {
  id: string;
  rank: number;
  title: string;
  author: string | null;
  publisher: string | null;
  category: string | null;
  sales_volume: number | null;
};

export type BestsellerTheme = {
  category: string;
  total_volume: number;
  book_count: number;
};

export type MarketInsights = {
  source: string;
  period_label: string | null;
  last_updated: string | null;
  books: BestsellerBook[];
  themes: BestsellerTheme[];
};
